Remove deprecated ImagePolicy status fields

cappyzawa opened the pull request:
Part of https://github.com/fluxcd/flux2/issues/5411

matheuscscp reviewed (approved):
LGTM! 🚀

matheuscscp commented:
Ah, forgot, we can also remove the `### Latest Image` and `### Observed Previous Image` sections in the API docs. Please update also the output of `kubectl get imagepolicy` in the `## Example` section according to the updated printer columns on the CRD:

```yaml
  - additionalPrinterColumns:
    - jsonPath: .status.latestRef.name
      name: Image
      type: string
    - jsonPath: .status.latestRef.tag
      name: Tag
      type: string
    - jsonPath: .status.conditions[?(@.type=="Ready")].status
      name: Ready
      type: string
    - jsonPath: .status.conditions[?(@.type=="Ready")].message
      name: Status
      type: string
    - jsonPath: .metadata.creationTimestamp
      name: Age
      type: date
```

cappyzawa commented:
I've addressed the review feedback with a separate commit (8352e23) for the documentation updates.

Note that the documentation changes are not directly related to the deprecated field removal itself - they're related to the printer columns changes that were implemented in #787. That's why I kept them as a separate commit rather than squashing with the main deprecation changes.
